felix/environment/feature_detect.go

```diff
@@ -165,6 +165,17 @@ func (d *FeatureDetector) refreshFeaturesLockHeld() {
 	}
 }
 
+// KernelIsAtLeast returns error is the predicates is either not true or running
+// kernel detection failed.
+func (d *FeatureDetector) KernelIsAtLeast(v string) error {
```

Comment:
Suggest adjusting the API of this and `isAtLeastKernel` to `bool, err` and then only returning an error if doing the check failed.  For your use case in tests, it makes a lot of sense to fail if checkin ghte kernel version failed.  for the original production code it made more sense to carry on and assume false.